scripts/deployment/tvm-deploy/tools/export.py

```diff
@@ -0,0 +1,188 @@
+
+                task = autotvm.task.create(func_create, args=tsk.args,
+                                           target=target, template_key='direct')
+                task.workload = tsk.workload
+
+                # create tuner
+                if tuner == 'xgb' or tuner == 'xgb-rank':
+                    tuner_obj = XGBTuner(task, loss_type='rank')
+                elif tuner == 'ga':
+                    tuner_obj = GATuner(task, pop_size=50)
+                elif tuner == 'random':
+                    tuner_obj = RandomTuner(task)
+                elif tuner == 'gridsearch':
+                    tuner_obj = GridSearchTuner(task)
+                else:
+                    raise ValueError("Invalid tuner: " + tuner)
+
+                # do tuning
+                n_trial=len(task.config_space)
+                tuner_obj.tune(n_trial=n_trial,
+                               early_stopping=early_stopping,
+                               measure_option=measure_option,
+                               callbacks=[
+                                   autotvm.callback.progress_bar(n_trial, prefix=prefix),
+                                   autotvm.callback.log_to_file(log_filename)])
+
+        #
+        tasks = autotvm.task.extract_from_program(sym, target=target,
+                                                  params=params, ops=(relay.op.nn.conv2d,))
+        logging.warning('Start tunning, this can be slow...')
+        tuning_option = {
+            'log_filename': 'tune.log',
+            'tuner': 'random',
+            'early_stopping': None,
+
+            'measure_option': autotvm.measure_option(
+                builder=autotvm.LocalBuilder(),
+                runner=autotvm.LocalRunner(number=10, repeat=1,
+                                           min_repeat_ms=1000),
+            ),
+        }
+        tune_kernels(tasks, **tuning_option)
+
+        with autotvm.apply_history_best(log_file):
+            with relay.build_config(opt_level=opt_level):
+                graph, lib, params = relay.build_module.build(
+                    sym, target=target, params=params)
+
+    else:
+        with relay.build_config(opt_level=opt_level):
+            graph, lib, params = relay.build(
+                sym, target, params=params)
+
+    # export library, json graph and parameters
+    lib.export_library(path + '_lib.so')
+    with open(path + '_graph.json', 'w') as fo:
+        fo.write(graph)
+    with open(path + '_{:04n}.params'.format(epoch), 'wb') as fo:
+        fo.write(relay.save_param_dict(params))
+        
+if __name__ == '__main__':
+    import gluoncv as gcv
+    if args.type == 'classify':
+        data_shape = (3, 224, 224)
+    else:
+        data_shape = (3, 512, 605)
+    model = gcv.model_zoo.get_model(args.model, pretrained=True)
+    export_tvm("model", model, data_shape, ctx=mx.cpu(), preprocess=TvmPreprocess(), target='llvm -mcpu=core-avx2', use_autotvm=False)
```

Comment:
can we add args for ctx and target?

Reply:
Sure, those variables can be set as arguments and assigned values according to specific environment. I just provide a demo for deployment using TVM.